internal/lib/stats/stats_server_linux.go

```diff
@@ -119,6 +150,100 @@ func (ss *StatsServer) populateNetworkUsage(stats *types.PodSandboxStats, sb *sa
+}
+
+// updatePodSandboxMetrics updates the sandbox metrics for the given sandbox.
+// If the sandbox is not found, it creates a new entry in the map.
+// Note: caller must hold the lock on the StatsServer.
+func (ss *StatsServer) updatePodSandboxMetrics(sb *sandbox.Sandbox) *SandboxMetrics {
+	if sb == nil {
+		return nil
+	}
+	sm, exists := ss.sboxMetrics[sb.ID()]
+	if !exists {
+		sm = NewSandboxMetrics(sb)
+	}
+	// network metrics are collected at pod level
+	if slices.Contains(ss.Config().IncludedPodMetrics, "network") {
+		podMetrics := ss.GenerateNetworkMetrics(sb)
+		sm.metric.Metrics = podMetrics
+	}
+
+	containersList := sb.Containers().List()
+	containerMetrics := make([]*types.ContainerMetrics, 0, len(containersList))
+	for _, c := range containersList {
+		// Skip if the container is stopped
+		if c.StateNoLock().Status == oci.ContainerStateStopped {
+			continue
+		}
+		cMetrics := ss.GenerateSandboxContainerMetrics(sb, c, sm)
+		containerMetrics = append(containerMetrics, cMetrics)
+	}
+	sm.metric.ContainerMetrics = containerMetrics
+	ss.sboxMetrics[sb.ID()] = sm // Update the entry in the map
+	return sm
+}
+
+// GenerateSandboxContainerMetrics generates a list of metrics for the specified sandbox
+// containers by collecting metrics from the cgroup based on the includedMetrics
+// except for network metrics which is collected at pod level.
+func (ss *StatsServer) GenerateSandboxContainerMetrics(sb *sandbox.Sandbox, c *oci.Container, sm *SandboxMetrics) *types.ContainerMetrics {
+	cgstats, err := ss.Runtime().ContainerStats(context.TODO(), c, sb.CgroupParent())
+	if err != nil || cgstats == nil {
+		logrus.Errorf("Error getting sandbox stats %s: %v", sb.ID(), err)
+		return nil
+	}
+	return ss.containerMetricsFromCgStats(sb, c, cgstats, sm)
+}
+
+func (ss *StatsServer) containerMetricsFromCgStats(sb *sandbox.Sandbox, c *oci.Container, cgstats *cgmgr.CgroupStats, sandboxMetrics *SandboxMetrics) *types.ContainerMetrics {
+	var metrics []*types.Metric
+	for _, m := range ss.Config().IncludedPodMetrics {
+		switch m {
+		case "cpu":
+			cpuMetrics := GenerateSandboxCPUMetrics(sb, c, cgstats.CPU)
+			metrics = append(metrics, cpuMetrics...)
+		case "memory":
+			memoryMetrics := GenerateSandboxMemoryMetrics(sb, c, cgstats.Memory, sandboxMetrics)
+			metrics = append(metrics, memoryMetrics...)
+		case "oom":
+			cm, err := ss.Config().CgroupManager().ContainerCgroupManager(sb.CgroupParent(), c.ID())
+			if err != nil {
+				logrus.Errorf("Unable to fetch cgroup manager %s: %v", sb.ID(), err)
+				continue
+			}
+			oomCount, err := cm.OOMKillCount()
+			if err != nil {
+				logrus.Errorf("Unable to fetch oom kill count %s: %v", sb.ID(), err)
+				continue
+			}
+			oomMetrics := GenerateSandboxOOMMetrics(sb, c, oomCount)
+			metrics = append(metrics, oomMetrics...)
+		case "network":
+			continue // network metrics are collected at pod level only
+		default:
+			logrus.Warnf("Unknown or misspelled metric: %s", m)
```

Comment:
Misspelled metrics are unknown metrics.

```suggestion
			logrus.Warnf("Unknown metric: %s", m)
```